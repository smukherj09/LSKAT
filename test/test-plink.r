library(LSKAT);

g.snp.hap1 <- "/ycga-ba/home/zw224/f/R/LSKAT/proj/simu/skat-test-1.hap"
g.snp.pos1 <- "/ycga-ba/home/zw224/f/R/LSKAT/proj/simu/skat-test-1.pos"

p <- longskat_gene_simulate(g.snp.hap1, g.snp.pos1, plink.format=T, file.plink.prefix="test-plink-lskat");

r.lskat1 <- longskat_gene_plink(p$file.plink.bed, p$file.plink.bim, p$file.plink.fam, p$file.gene.set, p$file.phe.long, p$file.phe.cov, 
								options=list(g.maxiter=6));
r.lskat1;
summary(r.lskat1);
plot( r.lskat1);
plot( r.lskat1, pdf.file="test-plink-lskat1.pdf", title="TEST PLINK 1", bonferroni=F);

r.lskat2 <- longskat_gene_plink(p$file.plink.bed, p$file.plink.bim, p$file.plink.fam, p$file.gene.set, p$file.phe.long, p$file.phe.cov, gene.range=c(1:5)
							    options=list(y.cov.time=2, g.maxiter=6));
r.lskat2;
summary(r.lskat2);
plot( r.lskat2);
plot( r.lskat2, pdf.file="test-plink-lskat2.pdf", title="TEST PLINK 2", bonferroni=T);

r.lskat3<-longskat_snp_plink( p$file.plink.bed, p$file.plink.bim, p$file.plink.fam, p$file.gene.set, p$file.phe.long, p$file.phe.cov, 
				    snp.range=c(1:199), 
				    options=list(y.cov.time=2, g.maxiter=6));
r.lskat3;
summary(r.lskat3);
plot( r.lskat3);
plot( r.lskat3, pdf.file="test-plink-lskat3.pdf", title="TEST PLINK 3", bonferroni=T);

unlink( c(p$file.plink.bed, p$file.plink.bim, p$file.plink.fam, p$file.gene.set, p$file.phe.long, p$file.phe.cov ) );