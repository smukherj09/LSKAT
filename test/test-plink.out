> library(LSKAT);
> 
> g.snp.hap1 <- "/ycga-ba/home/zw224/f/R/LSKAT/proj/simu/skat-test-1.hap"
> g.snp.pos1 <- "/ycga-ba/home/zw224/f/R/LSKAT/proj/simu/skat-test-1.pos"
> 
> p <- longskat_gene_simulate(g.snp.hap1, g.snp.pos1, plink.format=T, file.plink.prefix="test-lskat-plink");
MN.AR1--- -4.452133 4.718049 -0.0194428 0.6668912 1.44 0.8367757 
Writing FAM file to test-lskat-plink.fam 
Writing extended MAP file to test-lskat-plink.bim 
Writing BED file to test-lskat-plink.bed (Subject-major mode)
Genotype files have been converted into PLINK binary format(bed/bim/fam)
> 
> r.lskat <- longskat_gene_plink( p$file.plink.bed, p$file.plink.bim, p$file.plink.fam, 
+ 							    p$file.gene.set, p$file.phe.long, p$file.phe.cov, options=list(g.maxiter=6));
[ LONGSKAT_GENE_PLINK ] Procedure.
Checking the optional items......
* Covariate Count:  NA 
* Covariate Time Effect:  0 
* Parallel Computing:  No, 1 CPU(s)
* Debug Output:  No 
* C/C++ Module Used Output:  Yes 
* Beta Weights for Common SNPs:  0.5 0.5 
* Beta Weights for Rare SNPs:  1 25 
Checking PLINK file......
* BED file = test-lskat-plink.bed 
* BIM file = test-lskat-plink.bim 
* FAM file = test-lskat-plink.fam 
* Individuals = 800 SNPs= 53 
* PLINK loading successfully.
Checking phenotype file......
* PHE.LONG = test-lskat-plink-long.csv 
* PHE.TIME = 
Checking covariate file......
* COV.FILE = test-lskat-plink-cov.csv 
* Individuals = 800 
* COV.LEN = 2 
* COV.MEAN = -0.01915365 0.47875 
* COV.SD = 1.022367 0.4998607 
Checking gene definition file......
* GEN.SET.FILE = test-lskat-plink-gene.tab 
* GENEs = 1 
* SNPs = 53 
Starting to load all data files......
  PHE LONG = test-lskat-plink-long.csv 
* Individuals = 800 
* Times = 6 
* Mean = 1.42216 1.428647 1.451243 1.397834 1.355322 1.310522 
* SD = 1.373994 1.303784 1.37275 1.35957 1.374906 1.316606 
  PHE COV = test-lskat-plink-cov.csv 
* Individuals = 800 
* Covariate = 2 
* Mean = -0.01915365 0.47875 
* SD = 1.022367 0.4998607 
  GENO FAM = test-lskat-plink.fam 
* Individuals = 800 
  COMMON Individuals= 800 
Starting to estimate the SIGMA_A, SIGMA_B, SIGMA_E and other parameters......
* SIGMA_A = 0.4427947 
* SIGMA_B = 0.7683858 
* SIGMA_E = 0.8365135 
* RHO = 0.9829555 
* MU = 1.157094 
* Beta(Cov) = 0.5123904 0.5145688 
* Beta(Time) = 
* L(min) = 
* GENE.RANGE = 1 - 1 [ 1 ]
Starting the LSKAT estimate for each gene......
